src/session.cpp

```diff
@@ -399,9 +399,10 @@ class Impl : public dap::Session {
           return typeinfo->deserialize(d, data);
         })) {
       handlers.error("Failed to deserialize event '%s' body", event.c_str());
-      typeinfo->destruct(data);
-      delete[] data;
-      return {};
+      // "body" field in TerminatedEvent is an optional field.
```

Comment:
Please remove commented out lines.
If the field is optional, we shouldn't be raising errors if it's omitted.

I suggest:

```c++
    // "body" is an optional field for some events.
    bool body_ok = true;
    d->field("body", [&](dap::Deserializer* d) {
      if (!typeinfo->deserialize(d, data)) {
        body_ok = false;
      }
      return true;
    });

    if (!body_ok) {
      handlers.error("Failed to deserialize event '%s' body", event.c_str());
      typeinfo->destruct(data);
      delete[] data;
      return {};
    }
```